rsc/src/main/scala/rsc/semanticdb/Writer.scala

```diff
@@ -50,10 +50,22 @@ final class Writer private (
     }
     val converter = Converter(settings, reporter, gensyms, symtab, outline)
     val info = converter.toSymbolInformation
+
+    validate(outline, info)
+
     symbolBuf += info
     infos.put(outline.id.sym, info, outline.pos)
   }
 
+  private def validate(outline: Outline, info: s.SymbolInformation): Unit = {
+    (outline, info.signature) match {
+      case (defn: DefnMethod, m: s.MethodSignature) if m.returnType.isEmpty =>
```

Comment:
Where does rsc desugar `val x = ""` to a `DefnMethod`?

Reply:
In the scheduler, it will call `defnField` as step in `defnTemplate` since we need to look at each of its `stats`. `defnField` will call `assignSym` which generates a getter: https://github.com/twitter/rsc/blob/e57e96eb429e9e497103fdfcca5626ca7397799f/rsc/src/main/scala/rsc/outline/Scheduler.scala#L61